### DISCUSSION TITLE
updatePrimaryPalette and other styled mode theme utils on the server side (SSR)

### DISCUSSION THREAD

**Discussion by omnichronous** [primefaces/community#3195] [PrimeVue]
Hey everyone, I'm using PrimeVue 4 in Nuxt with SSR enabled and my use case led me to an interesting question.

I'm trying to make different pages in my app use different presets (really just a different primary color). I tried all the [utilities](https://primevue.org/theming/styled/#utils) - `usePreset`, `updatePreset`, `updatePrimaryPalette`, etc, and I tried them in many different places - the setup function, a composable, a plugin, a middleware, but the result is always the same - the page is always rendered on the server side with the original theme colors and the new preset is only applied only once the page loads on the client side. This causes a flash of unstyled content, of sorts.

Can anyone tell me what the "correct" way to manipulate the preset on the server side is?

**Comment by moraleja39** [comment]
I am facing a similar problem. In my case, I want to fetch some data from a server and update the preset based on it. Any hint on how to handle this would be appreciated.

**Reply by omnichronous** [reply]
Here's my workaround, maybe it helps you. I created a `useTheme` composable that uses `useHead`. My default theme is orange and with `useTheme('teal')` I can override it per-page.

```ts
export const useTheme = (theme: 'teal') => {
  if (theme === 'teal') {
    useHead({
      style: [
        {
          tagPosition: 'bodyOpen',
          id: 'my-theme',
          textContent: `:root {
  --p-primary-50: var(--p-teal-50);
  --p-primary-100: var(--p-teal-100);
  --p-primary-200: var(--p-teal-200);
  --p-primary-300: var(--p-teal-300);
  --p-primary-400: var(--p-teal-400);
  --p-primary-500: var(--p-teal-500);
  --p-primary-600: var(--p-teal-600);
  --p-primary-700: var(--p-teal-700);
  --p-primary-800: var(--p-teal-800);
  --p-primary-900: var(--p-teal-900);
  --p-primary-950: var(--p-teal-950);
}`,
        },
      ],
    })
  }
}
```
